README.md

```diff
@@ -20,23 +26,100 @@ chel migrate --from <backend> --to <backend> --out <dir-or-sqlitedb>
 
 Note: in many (if not all) instances, the `<url>` parameter can refer to a local folder path, in which case the command will operate without making a network connection, and will instead use the folder's contents to perform its operations.
 
-### `chel keygen`
+### `chel pin` - Per-Contract Versioning System
+
+🎯 Pin individual contracts to specific versions independently!
+
+**Key Features:**
+- ✅ **Per-contract versioning** using `chelonia.json` configuration
+- ✅ **Individual contract pinning** by specifying manifest file path
+- ✅ **New directory structure**: `contracts/<contract-name>/<version>/`
+- ✅ **Manifest-based workflow** - requires existing manifest files
+- ✅ **Ecosystem-agnostic** - no coupling to Node.js/npm
+
+**Workflow:**
+1. **Generate manifest first**: Use `chel manifest` to create `.manifest.json` files
+2. **Pin from manifest**: Use `chel pin` with the manifest file path
+3. **Contract files copied**: Only contract files (main/slim) are copied to new structure
+
+**Usage Examples:**
+```bash
+chel pin <version> <manifest-file-path>
+
+# Pin specific contract to a version using its manifest (from dist/contracts)
+chel pin 2.0.5 dist/contracts/2.0.5/chatroom.2.0.5.manifest.json
+chel pin 2.0.0 dist/contracts/2.0.0/group.2.0.0.manifest.json
 
+# Switch versions efficiently with --only-changed
+chel pin 2.0.6 dist/contracts/2.0.6/chatroom.2.0.6.manifest.json --only-changed
+
+# Note: Contracts are pinned to the contracts/ output directory
 ```
+
+**Configuration (`chelonia.json`):**
+```json
 {
-  "version": "1.0.0",
-  "cipher": "algo",
-  "pubkey": "...",
-  "privkey": "...",
-  "encrypted": "algo"
+  "contracts": {
+    "chatroom": {
+      "version": "2.0.6",
+      "path": "contracts/gi.contracts_chatroom/2.0.6/chatroom.2.0.6.manifest.json"
+    },
+    "group": {
+      "version": "2.0.0",
+      "path": "contracts/gi.contracts_group/2.0.0/group.2.0.0.manifest.json"
+    }
+  }
 }
 ```
 
-If `"encrypted"` doesn't exist - it means the `"privkey"` was saved in the clear.
+**Directory Structure Created:**
+```
+contracts/
+├── gi.contracts_chatroom/
+│   ├── 2.0.5/
+│   │   ├── chatroom.js
+│   │   └── chatroom-slim.js
+│   └── 2.0.6/
+│       ├── chatroom.js
+│       └── chatroom-slim.js
+└── gi.contracts_group/
+    └── 2.0.0/
+        ├── group.js
+        └── group-slim.js
+```
+
+> **Note:** The pin command only copies contract files (main and slim) from existing manifest files. It does not copy or generate manifest files - those should be created separately using `chel manifest`.
+
+> [!IMPORTANT]  
+> **Contract Manifest Files:**
+>
+> The `*.manifest.json` files generated by `chel pin` contain placeholder hashes and signatures for development purposes. For production deployment, you must generate proper cryptographic signatures using the following workflow:
```

Comment:
`chel pin` generates manifest files? What? o_O

Reply:
That's a mistake, it should be `chel manifest` not `chel pin`.

Reply:
Oh, I see. This section was added in initial workflow where `chel pin` created manifest file with placeholder hashes.

It's a mistake, will remove this section.